pkg/apis/common/validate.go

```diff
@@ -0,0 +1,32 @@
+package common
+
+import "errors"
+
+// ValidateTLSConfig validates a TLS configuration.
+func ValidateTLSConfig(tlsConfig *TLSConfig) error {
```

Comment:
do you actually need to validate it at all? won't it just fail if it is wrong?

Reply:
This validation is called by controller reconciliation not the real eventsource application, so it's needed.